clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/ingestion/heartbeat/HeartbeatMonitoringService.java

```diff
@@ -435,12 +435,14 @@ private class HeartbeatReporterThread extends Thread {
     @Override
     public void run() {
       while (!Thread.interrupted()) {
-        record();
         try {
+          record();
           TimeUnit.SECONDS.sleep(DEFAULT_REPORTER_THREAD_SLEEP_INTERVAL_SECONDS);
         } catch (InterruptedException e) {
           // We've received an interrupt which is to be expected, so we'll just leave the loop and log
           break;
+        } catch (Exception e) {
```

Comment:
Just one nit, how about potential errors. i.e. a Throwable but not an Exception, I guess we can let the thread to die, but at least we should log why it could happen.